sympy/polys/matrices/sdm.py

```diff
@@ -19,6 +19,33 @@ class SDM(dict):
     basic matrix arithmetic +, -, *, **.
     """
     def __init__(self, elemsdict, shape, domain):
+        """
+        Creates a :py:class:`~.SDM`.
+
+        Parameters
+        ==========
+
+        elemsdict : Represents non-zero elements of SDM as a dict of dicts
+        shape : Represents dimension of SDM
+        domain : Represents :py:class:`~.Domain` of SDM
+
+        Raises
+        ======
+
+        TypeError
+            If any of elemsdict, shape and domain are not provided
+
+        Examples
+        ========
+
+        >>> from sympy.polys.matrices.sdm import SDM
+        >>> from sympy import QQ
+        >>> elemsdict = {0:{1:QQ(2)}, 1:{}}
+        >>> A = SDM(elemsdict, (2, 2), QQ)
+        >>> A
+        {0: {1: 2}, 1: {}}
```

Comment:
The `elemsdict` should not have any empty dicts in there. This docstring should explain that it is the responsibility of whoever creates `elemsdict` to remove zero rows and zero elements.

Also this should show how to construct a `DomainMatrix` from the `SDM`.

Reply:
I think it would be better to move this docstring to be the main class docstring and also to show examples of using the `SDM` e.g. for adding and multiplying

Reply:
I'll do that